### PR TITLE
fix: update getVulnerabilities to match arborist api results

### PR DESCRIPTION
**Pull request by halcin**

### CODE REVIEW

**Review by PierreDemailly — APPROVED** [review]
lgtm

File: src/strategies/npm-audit.js [hunk 1]
```diff
@@ -21,9 +21,11 @@ async function getVulnerabilities(path, options = {}) {
   const arborist = new Arborist({ ...NPM_TOKEN, path });
 
   const { vulnerabilities } = (await arborist.audit()).toJSON();
+  const advisories = Object.values(vulnerabilities)
+    .flatMap((vuln) => (Array.isArray(vuln.via) && typeof vuln.via[0] === "object" ? vuln.via : []));
 
   if (useStandardFormat) {
-    return formatVulnerabilities(VULN_MODE.NPM_AUDIT, Object.values(vulnerabilities));
+    return formatVulnerabilities(VULN_MODE.NPM_AUDIT, Object.values(advisories));
```
Comment (tony-go) [inline src/strategies/npm-audit.js:28]: `flatMap`return an Array so we should probably not use `Object.values` here.

```suggestion
    return formatVulnerabilities(VULN_MODE.NPM_AUDIT, advisories);
```

**Review by tony-go — CHANGES_REQUESTED** [review]
Thanks for the PR 😍 

But we should add a unit test to cover this. cf https://github.com/NodeSecure/vulnera/tree/main/test/strategies/npm_audit

The error we found should have been caught somehow.

**Comment by halcin** [comment]
In fact the tests were already failing, I reported the problem. Antoine-coulon had proposed to solve them because it is necessary to rework the integration tests

**Review by fraxken — APPROVED** [review]

**Pull request merged**